app/components/submission/FileUploads/FileUploadsPage.js

```diff
@@ -16,42 +17,58 @@ const UPLOAD_MUTATION = gql`
   }
 `
 
-const FileUploadsPage = ({
-  manuscriptId,
-  setFieldValue,
-  errors,
-  touched,
-  values,
-  ...props
-}) => (
-  <Mutation mutation={UPLOAD_MUTATION}>
-    {(uploadFile, { loading, error: uploadError }) => {
-      const fieldName = 'files'
-      return (
-        <FileUploads
-          conversion={{
-            converting: loading,
-            // TODO import this constant from somewhere (data model package?)
-            completed: values[fieldName].includes(
-              file => file.type === 'MANUSCRIPT_SOURCE',
-            ),
-            error: uploadError,
-          }}
-          formError={errors[fieldName] && touched[fieldName]}
-          onDrop={([file]) =>
-            uploadFile({
-              variables: { file, id: values.id },
-            }).then(({ data }) => {
-              setFieldValue('title', data.uploadManuscript.title)
-              setFieldValue(fieldName, data.uploadManuscript.files)
-            })
-          }
-          setFieldValue={setFieldValue}
-          {...props}
-        />
-      )
-    }}
-  </Mutation>
-)
+class FileUploadsPage extends React.Component {
```

Comment:
Could you not create a common base class called "ComponentWithSaving" and then the FileUploadsPage and AuthorDetails page extend this base?

Reply:
I haven't seen much inheritance in Javascript before but I can't see why this would be bad in this case. This would solve my comment before https://github.com/elifesciences/elife-xpub/pull/175#issuecomment-395062980

Reply:
https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
I was just thinking about composition :)

Reply:
I think I'm just going to go ahead and wrap all the pages in another component which should have access to `props.values` then, and do this bit of logic there. If anybody objects say it now :)

Reply:
Ah HTML inheritance :D ... inheritance vs composition is very religious there are pluses and minuses on both sides. The article that @g-sam referenced is unfortunately more religious than helpful with its un-qualified assertions ...  This article is much better IMHO https://www.thoughtworks.com/insights/blog/composition-vs-inheritance-how-choose